python/shotgun_desktop/startup.py

```diff
@@ -775,6 +814,14 @@ def main(**kwargs):
             logger.info("Login canceled. Quitting.")
             return 0
 
+        # In the case where the site is using SSO, the user needs to renew
+        # its claims regularily. So we kick off a separate newewal thread.
+        if user._impl.get_saml_expiration() is not None:
```

Comment:
I agree with you guys that thing should be kept clean.

@manneohrstrom I just want to highlight once again that this code was done following a prior conversation with JF where it was decided that we (e.g. me) would implement an initial version of the renewal process purely to allow Francis to move ahead with the QA. It would also be a conversation-starter so that we could address that feature in a acceptable fashion.